internal/diag/diag.go

```diff
@@ -3,28 +3,91 @@ package diag
 import (
 	"fmt"
 	"go/ast"
+	"go/token"
+	"slices"
 
 	"golang.org/x/tools/go/analysis"
+
+	"github.com/manuelarte/funcorder/internal/astutils"
 )
 
-func NewConstructorNotAfterStructType(structSpec *ast.TypeSpec, constructor *ast.FuncDecl) analysis.Diagnostic {
+func NewConstructorNotAfterStructType(fset *token.FileSet,
+	structSpec *ast.TypeSpec, constructor *ast.FuncDecl,
+) (analysis.Diagnostic, error) {
+	suggestedFixConstructorByte, err := astutils.NodeToByteArray(fset, constructor)
```

Comment:
I propose to do a few renamings.
`[]byte` is a slice, not an array.

```suggestion
	suggestedBytes, err := astutils.NodeToBytes(fset, constructor)
```